packages/storage/src/providers/s3/utils/isValidObjectUrl.ts

```diff
@@ -0,0 +1,25 @@
+import { extendedEncodeURIComponent } from '@aws-amplify/core/internals/aws-client-utils';
```

Comment:
Need to add the license header to this file

```suggestion
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { extendedEncodeURIComponent } from '@aws-amplify/core/internals/aws-client-utils';
```

Reply:
updated and reworked the check to be a void function and throw on error